datahub-web-react/src/app/ingestV2/source/IngestionSourceList.tsx

```diff
@@ -452,6 +470,25 @@ export const IngestionSourceList = ({
         }
     };
 
+    // Handle executing or refetching of a created or updated source via the separated page
+    const [isCreatedOrUpdatedSourceFromLocationHandled, setIsCreatedOrUpdatedSourceFromLocationHandled] =
+        useState<boolean>(false);
+    useEffect(() => {
+        if (createdOrUpdatedSourceUrnFromLocation && !isCreatedOrUpdatedSourceFromLocationHandled) {
+            setIsCreatedOrUpdatedSourceFromLocationHandled(true);
+            if (shouldRunCreatedOrUpdatedSourceFromLocation) {
+                executeIngestionSource(createdOrUpdatedSourceUrnFromLocation);
+            } else {
+                setSourcesToRefetch((prev) => new Set([...prev, createdOrUpdatedSourceUrnFromLocation]));
```

Comment:
the way this works is that we fetch once and determine if we want to keep refetching right? like just cuz we updated a source doesn't mean we continue to keep refetching (unless we should because it's in the middle of a run) right?